mods/default/nodes.lua

```diff
@@ -1613,6 +1617,96 @@ minetest.register_node("default:bush_sapling", {
 	end,
 })
 
+minetest.register_node("default:huckleberry_bush_stem", {
+	description = "Huckleberry Bush Stem",
+	drawtype = "plantlike",
+	visual_scale = 1.41,
+	tiles = {"default_huckleberry_bush_stem.png"},
+	inventory_image = "default_huckleberry_bush_stem.png",
+	wield_image = "default_huckleberry_bush_stem.png",
+	paramtype = "light",
+	sunlight_propagates = true,
+	groups = {choppy = 2, oddly_breakable_by_hand = 1, flammable = 2},
+	sounds = default.node_sound_wood_defaults(),
+	selection_box = {
+		type = "fixed",
+		fixed = {-7 / 16, -0.5, -7 / 16, 7 / 16, 0.5, 7 / 16},
+	},
+})
+
+minetest.register_node("default:huckleberry_bush_leaves", {
+	description = "Huckleberry Bush Leaves",
+	drawtype = "allfaces_optional",
+	waving = 1,
+	tiles = {"default_huckleberry_bush_leaves.png"},
+	paramtype = "light",
+	groups = {snappy = 3, flammable = 2, leaves = 1},
+	drop = {
+		max_items = 1,
+		items = {
+			{items = {"default:huckleberry_bush_sapling"}, rarity = 5},
+			{items = {"default:huckleberries"}}
+		}
+	},
+	sounds = default.node_sound_leaves_defaults(),
+
+	after_place_node = default.after_place_leaves,
+})
+
+minetest.register_node("default:huckleberry_bush_sapling", {
+	description = "Huckleberry Bush Sapling",
+	drawtype = "plantlike",
+	tiles = {"default_huckleberry_bush_sapling.png"},
+	inventory_image = "default_huckleberry_bush_sapling.png",
+	wield_image = "default_huckleberry_bush_sapling.png",
+	paramtype = "light",
+	sunlight_propagates = true,
+	walkable = false,
+	on_timer = default.grow_sapling,
+	selection_box = {
+		type = "fixed",
+		fixed = {-4 / 16, -0.5, -4 / 16, 4 / 16, 2 / 16, 4 / 16}
+	},
+	groups = {snappy = 2, dig_immediate = 3, flammable = 2,
+		attached_node = 1, sapling = 1},
+	sounds = default.node_sound_leaves_defaults(),
+
+	on_construct = function(pos)
+		minetest.get_node_timer(pos):start(math.random(300, 1500))
+	end,
+
+	on_place = function(itemstack, placer, pointed_thing)
+		itemstack = default.sapling_on_place(itemstack, placer, pointed_thing,
+			"default:huckleberry_bush_sapling",
+			-- minp, maxp to be checked, relative to sapling pos
+			{x = -1, y = 0, z = -1},
+			{x = 1, y = 1, z = 1},
+			-- maximum interval of interior volume check
+			2)
+
+		return itemstack
+	end,
+})
+
+minetest.register_node("default:huckleberries", {
```

Comment:
Apples are only nodes because that is how they exist in the world, picked berries should probably be items.